homeassistant/helpers/entity_component.py

```diff
@@ -126,40 +180,56 @@ def add_entity(self, entity, platform=None):
                 object_id = '{} {}'.format(platform.entity_namespace,
                                            object_id)
 
-            entity.entity_id = generate_entity_id(
+            entity.entity_id = async_generate_entity_id(
                 self.entity_id_format, object_id,
                 self.entities.keys())
 
         self.entities[entity.entity_id] = entity
-        entity.update_ha_state()
+        self.hass.loop.create_task(entity.async_update_ha_state())
```

Comment:
Also all other invocations of `async_add_entities` will have to be yielding or using `create_task` if they don't care about the result.